assets/js/src/core/modules/element/dynamic-types/definitions/objects/data-related/components/many-to-many-relation/hooks/use-value.ts

```diff
@@ -33,22 +38,47 @@ interface UseValueReturn {
   addItems: (items: ManyToManyRelationValueItem[]) => void
   addAssets: (assets: Asset[]) => Promise<void>
   maxRemainingItems?: number
+  pathFormatterClass?: string | null
 }
 
 export const useValue = (
   value: ManyToManyRelationValue | null,
   setValue: (value: ManyToManyRelationValue | null) => void,
   displayedValue: ManyToManyRelationValue | null,
   setDisplayedValue: (value: ManyToManyRelationValue | null) => void,
+  setIsLoading: (isLoading: boolean) => void,
   maxItems: number | null,
-  allowMultipleAssignments?: boolean
+  id: string | undefined,
+  allowMultipleAssignments?: boolean,
+  pathFormatterClass?: string | null
 ): UseValueReturn => {
+  const { id: dataObjectId } = useDataObject()
+  const { formatPath } = useDataObjectHelper()
   const modal = useAlertModal()
+
   const { t } = useTranslation()
   const itemIsInValue = (id: number, type: string): boolean => {
     return value?.some(item => item.id === id && item.type === type) ?? false
   }
 
+  useEffect(() => {
+    if (pathFormatterClass !== null && value !== null && dataObjectId !== undefined && id !== undefined) {
+      setIsLoading(true)
+
+      formatPath(value, id, dataObjectId).then((data) => {
+        if (data.items !== undefined) {
+          const newValue = value.map((item) => ({
+            ...item,
+            fullPath: data.items.find(i => i.objectReference === `object_${item.id}`)?.formatedPath ?? item.fullPath
+          }))
+
+          setDisplayedValue(newValue)
+          setIsLoading(false)
+        }
+      }).catch(error => { trackError(new ApiError(error as unknown as ApiErrorData)) })
```

Comment:
we should use `trackError(new ApiError())` only when we are sure that the error is an API error, therefore please move the logic to the other suggested location and do just a console.error here if we have other errors.